recipes/full_finetune_distributed.py

```diff
@@ -29,6 +29,8 @@
 
 log = utils.get_logger("DEBUG")
 
+os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
```

Comment:
I think that it should be a parameter in the config, but we should not expose it. The default is true.

If some user faces an issue, they can just turn it off with a flag, instead of having to create a new recipe. Something like:

```
# Expandable segments save reserved memory and cant prevent OOM.
# If you have an issue because of this flag, you can set it to false in your config or add a flag to
# you command `tune run <your configurations> use_expandable_segments=False`
# For more info, check: <link to the PR or issue>
use_expandable_segments = cfg.get("use_expandable_segments", "True)
if use_expandable_segments:
	os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
```

I have just tested qwen/gemma/llama on lora/full single/dist and it worked well for the most part. Just some warnings with qwen/gemma.